recirq/qcqmc/hamiltonian.py

```diff
@@ -0,0 +1,319 @@
+
+
+@attrs.frozen
+class LoadFromFileHamiltonianParams(Params):
+    """Class for storing the parameters that specify loading integrals from a file."""
+
+    name: str
+    integral_key: str
+    n_orb: int
+    n_elec: int
+    do_eri_restore: bool = False
+    path_prefix: str = ""
+
+    def _json_dict_(self):
+        return attrs.asdict(self)
+
+    @property
+    def path_string(self, prefix: str = "") -> str:
+        return self.path_prefix + OUTDIRS.DEFAULT_HAMILTONIAN_DIRECTORY + self.name
+
+
+# xyz coordinates of an atom
+PositionT = Tuple[float, float, float]
+# [atom name, xyz]
+GeomT = Tuple[str, PositionT]
+
+
+@attrs.frozen
+class PyscfHamiltonianParams(Params):
+    """Class for describing the parameters to get a Hamiltonian from Pyscf.
+
+    name: A name for the Hamiltonian
+    n_orb: The number of orbitals (redundant with later parameters but used
+        for validation).
+    n_elec: The number of electrons (redundant with later parameters but
+        used for validation).
+    geometry: The coordinates of each atom. An example is [('H', (0, 0, 0)),
+        ('H', (0, 0, 0.7414))]. Distances are in angstrom. Use atomic
+        symbols to specify atoms.
+    basis: The basis set, e.g., 'cc-pvtz'.
+    multiplicity: The spin multiplicity
+    charge: The total molecular charge.
+    rhf: Whether to use RHF (True) or ROHF (False).
+    verbose_scf: Setting passed to pyscf's scf verbose attribute.
+    save_chkfile: If True, then pyscf will save a chk file for this Hamiltonian.
+    overwrite_chk_file: If save_chkfile and overwrite_chk_file are both true then Pyscf
+        will be allowed to overwrite the previously saved chk file. Otherwise, if save_chkfile
+        is True and overwrite_chk_file is False, then we raise a FileExistsError if the
+        chk file already exists.
+    path_prefix: A prefix to prepend to the path where the data will be saved (if any)
+    """
+
+    name: str
+    n_orb: int
+    n_elec: int
+    geometry: Tuple[GeomT, ...] = attrs.field(
+        converter=lambda x: tuple((at[0], tuple(at[1])) for at in x)
+    )
+    basis: str
+    multiplicity: int
+    charge: int = 0
+    rhf: bool = True
+    verbose_scf: int = 0
+    save_chkfile: bool = False
+    overwrite_chk_file: bool = False
+    path_prefix: str = ""
+
+    def _json_dict_(self):
+        return attrs.asdict(self)
+
+    @property
+    def path_string(self) -> str:
+        """Output path string"""
+        return self.path_prefix + OUTDIRS.DEFAULT_HAMILTONIAN_DIRECTORY + self.name
+
+    @property
+    def chk_path(self) -> Path:
+        """Path string to any checkpoint file is saved."""
+        parent = self.base_path.parent
+        if not parent.is_dir():
+            parent.mkdir(exist_ok=True, parents=True)
+        return self.base_path.with_suffix(".chk")
+
+    @property
+    def pyscf_molecule(self) -> gto.Mole:
+        """Underlying pyscf mol instance."""
```

Comment:
mol -> molecule.  Avoid abbreviations, especially when they are ambiguous.

Reply:
I think it's pyscf doing the abbreviating here, but they seem to call it `Mole`

Reply:
Is that the animal or the sauce? I'll update to Mole

Reply:
Done.